shared/ui/core/src/test/java/com/ivy/ui/FormatMoneyUseCaseTest.kt

```diff
@@ -23,33 +23,80 @@ class FormatMoneyUseCaseTest {
     enum class MoneyFormatterTestCase(
         val amount: Double,
         val showDecimal: Boolean,
+        val shortenAmount: Boolean,
         val locale: Locale,
         val expectedOutput: String
     ) {
         ENG_SHOW_DECIMAL(
-            amount = 1000.12,
+            amount = 1_000.12,
             showDecimal = true,
+            shortenAmount = false,
             locale = Locale.ENGLISH,
             expectedOutput = "1,000.12"
         ),
         ENG_HIDE_DECIMAL(
-            amount = 1000.12,
+            amount = 1_000.12,
             showDecimal = false,
+            shortenAmount = false,
             locale = Locale.ENGLISH,
             expectedOutput = "1,000"
         ),
         GERMAN_SHOW_DECIMAL(
-            amount = 1000.12,
+            amount = 1_000.12,
             showDecimal = true,
+            shortenAmount = false,
             locale = Locale.GERMAN,
             expectedOutput = "1.000,12"
         ),
         GERMAN_HIDE_DECIMAL(
-            amount = 1000.12,
+            amount = 1_000.12,
             showDecimal = false,
+            shortenAmount = false,
             locale = Locale.GERMAN,
             expectedOutput = "1.000"
         ),
+        ENGLISH_1K_SHORT_AMT(
+            amount = 13_000.10,
+            showDecimal = true,
+            shortenAmount = true,
+            locale = Locale.ENGLISH,
+            expectedOutput = "13.00k"
+        ),
+        ENGLISH_MILLION_SHORT_AMT(
+            amount = 1_230_000.10,
+            showDecimal = true,
+            shortenAmount = true,
+            locale = Locale.ENGLISH,
+            expectedOutput = "1.23m"
+        ),
+        ENGLISH_BILLION_SHORT_AMT(
+            amount = 1_233_000_000.10,
+            showDecimal = true,
+            shortenAmount = true,
+            locale = Locale.ENGLISH,
+            expectedOutput = "1.23b"
+        ),
+        GERMAN_1K_SHORT_AMT(
+            amount = 13_000.10,
+            showDecimal = true,
+            shortenAmount = true,
+            locale = Locale.GERMAN,
+            expectedOutput = "13,00k"
```

Comment:
That's not okay, it should be just "13k", use decimal format for the shortening amounts to handle such cases

Reply:
using same decimal formatting now for large numbers too so, it will be 13k only when `showDecimal `is false otherwise it'd contain decimal places. Let me know if you'd still prefer to skip decimal places regardless of the value of   `showDecimal `

Reply:
@shamim-emon let's skip decimal places regardless, the purposes of "13k" is to shorter "13,000" and "13.00k" is the same length and looks weird

Reply:
@ILIYANGERMANOV in that case, what would be expected output of **1_233_500.10** ? 
i). 1m
ii). 1.23m
asking this cause if it's i then we can use one of the existing formatter of the class. But if it's ii. will need to add some additional logic.
Please let me know your comment on this.

Reply:
1.23m ofc, we're just making sure that unnecessary zeros are not showing - to achieve this use this `DecimalFormatter("###,###.##")`. I.e. your usecase should mirror the behavior of the existing logic

Reply:
For simplicity, you can just use the formatter that I wrote above for the case of shortening